scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py

```diff
@@ -1860,6 +1860,50 @@ def test_guidelines_link_active(self):
 class TestClientListMethodsUseCorePaging(pylint.testutils.CheckerTestCase):
     CHECKER_CLASS = checker.ClientListMethodsUseCorePaging
 
+    def test_list_return_type_acceptable(self):
+        class_node, function_node = astroid.extract_node("""
+        from azure.core.paging import ItemPaged
+
+        class SomeClient(): #@
+            def _list_thing(self): #@
```

Comment:
should this have the underscore prefix? I think the linter rule will ignore if it does, right?

Reply:
Right now the rule just checks if the function is a method, node.is_method(). I will make it ignore private functions